src/main/webapp/cluster-view/cluster-view.controller.js

```diff
@@ -488,6 +488,54 @@ var app = angular.module('managementConsole')
         return cachesInput;
       }
     };
+  }).filter('cacheStatus', function (modelController, clusterNodesService) {
+    var cachedIsRebalancing = {};
+    var cachedIsSplitBrain = {};
+
+    function cacheStatusFilter(cachesInput, scope) {
+      var checkboxes = scope.statusCheckboxes;
+      var atLeastOneFilterOn = checkboxes.indexing || checkboxes.offline || checkboxes.rebalancing || checkboxes.splitbrain;
+      if (atLeastOneFilterOn) {
+        var caches = [];
+        angular.forEach(cachesInput, function (cache) {
+          if (checkboxes.rebalancing) {
+            if (cache.name in cachedIsRebalancing) {
+              var rebalancing = typeof cachedIsRebalancing[cache.name] === 'boolean' ? cachedIsRebalancing[cache.name] : undefined;
+              if (rebalancing) {
+                caches.push(cache);
+              }
+            } else {
+              modelController.getServer().fetchCacheStats(scope.currentCluster, cache).then(function (response) {
+                var stats = response[0];
+                cachedIsRebalancing[input] = stats['cache-rebalancing-status'] === 'IN_PROGRESS' ? true : false;
```

Comment:
@vblagoje `input` is undefined. Is this meant to be `cachedIsRebalancing[cache.name] = ...` ?